submariner/multiple-cable-drivers.md

```diff
@@ -0,0 +1,415 @@
+    Spec ClusterConnectionPolicy `json:"spec"`
+}
+
+type ClusterConnectionPolicySpec struct {
+    // LeftClusterSelector identifies the Cluster resources representing the clusters on one end of a connection. An empty
+    // selector indicates wildcard, ie matches any cluster.
+    // Not used in combination with ClusterSelector.
+    LeftClusterSelector metav1.LabelSelector `json:"leftClusterSelector,omitempty"`
+
+    // RightClusterSelector identifies the Cluster resources representing the clusters on the other end of a connection.
+    // An empty selector indicates wildcard, ie matches any cluster.
+    // Not used in combination with ClusterSelector.
+    RightClusterSelector metav1.LabelSelector `json:"rightClusterSelector,omitempty"`
+
+    // ClusterSelector identifies the Cluster resources representing a subset of clusters.
+    // Not used in combination with LeftClusterSelector or RightClusterSelector.
+    // An empty selector indicates wildcard, ie matches any cluster.
+    ClusterSelector metav1.LabelSelector `json:"clusterSelector,omitempty"`
+
+    // Topology identifies the connection topology for the cable ends: full-mesh/point-to-point/client-server.
+    Topology `json:"topology,omitempty"`
+
+   // Priority identifies the priority for the connection policy. Higher priorities are applied first.
+    Priority  *int  `json:"priority,omitempty"`
+
+    // ConnectionPolicy identifies the default connection policy if no policies are found: to connect or no-connect.
+    ConnectionPolicy *string `json:"connectionPolicy,omitempty"`
+
+    Spec CableDriverSpec `json:"spec"`
+}
+
+type CableDriverSpec struct {
+    // Name of the cable driver implementation.
+    Name string `json:"name"`
+
+    // Options specifies key/value configuration parameters for the cable driver.
+    Options *v1.ConfigMap `json:"options,omitempty"`
+}
+```
+
+The `ClusterConnectionPolicies` CR defines the criteria for selecting which cluster pairs/subsets can connect and associated optional
+configuration options to use for each cluster connection. To resolve the connectivity policy for a remote cluster's Endpoint,
+Submariner would:  
+
+- Search for a policy whose left and right label selectors match the corresponding Cluster resources of the local
+and remote Endpoints.
+- Search for a policy whose cluster label selector match the corresponding Cluster resources of the local and remote Endpoints.
+
+If any matching policy is found, the clusters are either:
+
+- connected (based on the connectionPolicy) according to the specified topology in the policy. Or
+- not connected (based on the connectionPolicy).
+
+If more than one matching policy is found, the priority field is used to determine which policy to use. If policies of equal priority are found
+it's assumed that these policies are to be applied in parallel.
+
+If no matching policy is found, the default cable driver and default connection policy are used. This policy can be auto-created at broker
```

Comment:
The default cable driver should be the one defined on the `Submariner` CR

Reply:
Yes